python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -400,7 +406,11 @@ class RandomForestRegressor(Base):
         cdef  RandomForestMetaData[double, double] *rf_forest64 = \
             new RandomForestMetaData[double, double]()
 
-        if self.dtype == np.float32:
+        self.file_name = state["file_name"]
```

Comment:
it'd be better to generate a new filename when needed instead of storing it in state as I mentioned above I think